Sources/SwiftDocC/LinkTargets/LinkDestinationSummary.swift

```diff
@@ -219,14 +219,18 @@ public extension DocumentationNode {
             LinkDestinationSummary(landmark: $0, basePath: presentationURL.path, page: self, platforms: platforms, compiler: &compiler)
         }
         
+        var variantTaskGroups: [[RenderNode.Variant.Trait]: [LinkDestinationSummary.TaskGroup]] = [:]
         let taskGroups: [LinkDestinationSummary.TaskGroup]
         switch kind {
         case .tutorial, .tutorialArticle, .technology, .technologyOverview, .chapter, .volume, .onPageLandmark:
             taskGroups = [.init(title: nil, identifiers: context.children(of: reference).map { $0.reference.absoluteString })]
         default:
             taskGroups = renderNode.topicSections.map { group in .init(title: group.title, identifiers: group.identifiers) }
+            for variant in renderNode.topicSectionsVariants.variants {
+                variantTaskGroups[variant.traits] = variant.applyingPatchTo(renderNode.topicSections).map { group in .init(title: group.title, identifiers: group.identifiers) }
```

Comment:
Yes, I think that will have a performance impact. When linkable entities are emits, this code runs would run for every page and landmark that have topic section variants. In the worst case this could end up encoding all the pages.